✨ Add support for Asana Project Creation

awitkowski0 opened the pull request:
Added New Options within the Asana Node.

Namely,
 - Project Creation
 - Project Deletion
 - Project Updates
 
![image](https://user-images.githubusercontent.com/26530277/145072854-85499187-0b2a-4d34-9fdf-fefc0e85c83b.png)

 This is my first PR to this repo, I have viewed all the necessary guidelines and code of conduct, but if I am in violation of anything or need to do anything else please let me know! Thanks for creating this wonderful product.

awitkowski0 committed:
✨ Add support for Asana Project Creation

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/n8n-io/n8n?pullRequest=2532) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/n8n-io/n8n?pullRequest=2532) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/n8n-io/n8n?pullRequest=2532) it.</sub>

RicardoE105 commented on packages/nodes-base/nodes/Asana/Asana.node.ts:
```diff
@@ -2197,6 +2433,48 @@ export class Asana implements INodeType {
+
+						// required parameters
+						body.name = this.getNodeParameter('name', i);
+						body.workspace = this.getNodeParameter('workspace', i);
```
The API does not need a workspace parameter.

RicardoE105 commented on packages/nodes-base/nodes/Asana/Asana.node.ts:
```diff
@@ -1493,11 +1494,142 @@ export class Asana implements INodeType {
 						description: 'Get all projects',
 					},
+					{
+						name: 'Update Project',
```
Should be called "Update"

RicardoE105 commented on packages/nodes-base/nodes/Asana/Asana.node.ts:
```diff
@@ -1625,6 +1757,111 @@ export class Asana implements INodeType {
+				description: 'The project to update info on.',
+			},
+			{
+				displayName: 'Additional Fields',
```
Should be called update fields

RicardoE105 commented on packages/nodes-base/nodes/Asana/Asana.node.ts:
```diff
@@ -1625,6 +1757,111 @@ export class Asana implements INodeType {
+			{
+				displayName: 'Project',
+				name: 'projectId',
+				type: 'options',
```
Project is type options but does not have any options?

awitkowski0 replied:
Originally had it as getProjects but it is not as reliable as using a string so I switched to a string option

RicardoE105 commented on packages/nodes-base/nodes/Asana/Asana.node.ts:
```diff
@@ -1625,6 +1757,111 @@ export class Asana implements INodeType {
+						name: 'team',
+						type: 'options',
+						typeOptions: {
+							loadOptionsMethod: 'getTeams',
```
Add load dependency. Otherwise, if the workspace is not set, this will not work.

awitkowski0 replied:
Added a load dependency and I added workspace as a general field for it with a getWorkspaces option for it

RicardoE105 commented on packages/nodes-base/nodes/Asana/Asana.node.ts:
```diff
@@ -2248,6 +2526,45 @@ export class Asana implements INodeType {
+							body.team = additionalFields.team;
+						}
+						if (additionalFields.workspace) {
+							body.workspace = additionalFields.workspace;
```
Workspace it;s not needed to update a project.

awitkowski0 replied:
Fixed in latest commit

RicardoE105 commented:
@awitkowski0 thanks very much for your contribution. Great to hear that you are enjoying n8n. Can you please address the comment in my review? Let me know if you have questions.

awitkowski0 committed:
Merge branch 'n8n-io:master' into master

awitkowski0 committed:
Renamed Additional Field to Update Fields
Removed Workspace from Update Fields
Renamed Update Project to Update
Changed Project Type options under update to string for name
Added Workspace as a requirement to update projects and as a load dependency
Removed Workspace from actual updating

awitkowski0 commented:
Committed again with a bunch of changes made to resolve your questions @RicardoE105. Hopefully everything is resolved! Sorry for taking so long between commits, a lot of stuff came up in between.

janober commented:
Thanks a lot for your contribution. Sorry that it took so long. Got merged and will be released with the next version.

janober commented:
Got released with `n8n@0.165.0`
